### PR TITLE
Add `isRelevantForPreferred` field to external attribution sources

### PR DESCRIPTION
**Pull request by MarkusObendrauf**
### Summary of changes

Add new `isRelevantForPreferred` field to external attribution sources. This field has type optional boolean. If set, it indicates that signals from this source may be 'preferred over' using the preference feature.

If no sources have this field set, then the preference feature is disabled.

### Context and reason for change

We would like for the preference feature to only apply to signals from certain sources.

### How can the changes be tested

See unit tests

I have updated the example files `example.opossum` & `opossum_input.json` to include relevant sources. The preferred feature should be enabled on those files. It should be disabled on every other random file you have.

### CODE REVIEW

File: src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts [hunk 1]
```diff
@@ -234,10 +235,12 @@ export function getMergeButtonsDisplayState(currentState: {
         currentState.packageInfoWereModified) ||
       currentState.targetAttributionIsPreSelected,
     hideMarkAsPreferredButton:
+      !currentState.isPreferenceFeatureEnabled ||
```
Comment (FlorianSchepers) [inline src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts:238]: Optional: A integration test that checks if the button is hidden if the preference Feature is enables, i.e., by using get by text.

**Review by FlorianSchepers — APPROVED** [review]
lgtm 

- see one comment in code
- squash commits
- add proper commit message

File: src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts (hunk 1 above)
Comment (MarkusObendrauf) [reply src/Frontend/Components/AttributionColumn/attribution-column-helpers.ts:238]: good idea! added one

**Comment by MarkusObendrauf** [comment]
squashed and rebased

**Pull request merged**